src/mono/mono/metadata/icall.c

```diff
@@ -958,9 +958,17 @@ void
 ves_icall_System_Buffer_BulkMoveWithWriteBarrier (guint8 *destination, guint8 *source, size_t len, MonoType *type)
```

Comment:
I think this method is always called when type contains references. We can probably discard the `type` argument and have a new GC api method, something like `mono_gc_move_with_write_barrier` that ends up calling `wbarrier_value_copy`. `mono_gc_wbarrier_range_copy` could also be reused.

Reply:
@BrzVlad, thanks for the idea. I am not sure if it is possible to drop the `type` argument though. It looks like the function `mono_gc_wbarrier_value_copy_internal` relies critically on the `klass` to get GC description for `scan_object_for_binary_protocol_copy_wbarrier`, am I missing something?